src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,456 @@
+   * @returns {Promise<Array>} - An array of transfer transactions for the specified address.
+   * @throws {Error} - If an unknown error occurs during the request.
+   */
+  async function getTokensHistory({ recent, address }: { recent: boolean, address: string }) {
+    const addressTransactions = transactions.value[address]?.transactions || [];
+
+    if (addressTransactions?.length && !recent) {
+      return addressTransactions;
+    }
+
+    let rawTransactions = [];
+    const lastTransaction = addressTransactions?.[0];
+
+    if (recent) {
+      let nextPageUrl = null;
+      let isAllNewTransactionsLoaded = false;
+
+      await watchUntilTruthy(() => middleware.value);
+
+      while (!isAllNewTransactionsLoaded && nextPageUrl !== null) {
+        const url = nextPageUrl
+          ? `${activeNetwork.value.middlewareUrl}/${nextPageUrl}`
+          : `${activeNetwork.value.middlewareUrl}/v2/aex9/transfers/to/${address}`;
+        // eslint-disable-next-line no-await-in-loop
+        const { data, next }: any = await fetchJson(url);
+
+        if (!data?.length) break;
+
+        rawTransactions.push(...data);
+
+        if (
+          data.some((t: any) => t?.tx_hash === lastTransaction?.hash)
+          || !addressTransactions?.length
+        ) {
+          isAllNewTransactionsLoaded = true;
+        }
+
+        nextPageUrl = next;
+      }
+
+      if (rawTransactions?.[0]?.tx_hash === lastTransaction?.hash) {
+        return addressTransactions.slice(0, 10);
+      }
+    } else {
+      rawTransactions = await fetchJson(`${activeNetwork.value.middlewareUrl}/aex9/transfers/to/${address}`);
+    }
+
+    const newTransactions = rawTransactions
+      .filter((tx: any) => !getDexContracts.value.router.includes(tx.contract_id))
+      .map((tx: any) => ({
+        ...tx,
+        tx: {
+          contractId: tx.contract_id,
+          senderId: tx.sender,
+          recipientId: tx.recipient,
+          callerId: tx.sender,
+          type: 'ContractCallTx',
+        },
+        recipient: tx.recipient,
+        incomplete: true,
+        microTime: tx.micro_time,
+        hash: tx.tx_hash,
+      }));
+
+    const newLocalTransactions = { ...transactions.value };
+    const addressData = newLocalTransactions[address] || { transactions: [] };
+
+    if (newTransactions?.[0]?.hash !== lastTransaction?.hash && recent) {
+      addressData.transactions = uniqBy([...newTransactions, ...addressData.transactions], 'hash');
+    } else {
+      addressData.transactions = newTransactions.reverse();
+    }
+
+    newLocalTransactions[address] = addressData;
+
+    transactions.value = { ...newLocalTransactions };
+
+    return addressData.transactions;
+  }
+
+  function getTxSymbol(tx?: ITx): string {
+    if (!tx) {
+      return AETERNITY_SYMBOL;
+    }
+    if (tx?.contractId) {
+      return availableTokens.value[tx.contractId]?.symbol;
+    }
+    const contractCallData = categorizeContractCallTxObject({ tx });
```

Comment:
This will return incorrect results in several cases, the parameter suppose to be a proper transaction not just an inner one. The fields `incomplete`, `pending` and `amount` are also required.